content/articles/object-serialization-and-deserialization-in-java/index.md

```diff
@@ -0,0 +1,351 @@
+- During the serialization process, only preserve non-static data members; static and temporary data members are not.
+- The String and its wrapper classes have the `Serializable` interface implemented by default.
+
+### How to serialize and deserialize with the help of examples
+
+To perform serialization, use the `ObjectOutputStream` class's `writeObject` method. For deserialization, use the `InputObjectStream` class's `readObject` method.
+> The `.ser` extension is usually used when serializing an entity to a directory in Java.
+
+Method syntax of writeObject:
+
+```java
+public final void writeObject(Object o) throws IOException
+```
+
+Method syntax readObject:
+
+```java
+public final Object readObject() throws IOException, ClassNotFoundException
+```
+
+> We are going to use sections of our full example code which will be shown later. Below are the sections of the full code. 
+
+#### A Serialization example
+
+Consider this, `Link` class, which contains two data members: `commons` and `favorites`. This is how we will define it:
+
+```java
+import java.io.*;
+
+class Link implements Serializable {
+
+    private String commons;
+    protected int favorites;
+
+    public Link(String commons, int favorites) {
+        this.commons = commons;
+        this.favorites = favorites;
+    }
+
+    public void printLink() {
+        System.out.println("Link : " + this.commons);
+    }
+
+}
+```
+
+To serialize this class, we first need to implement the `java.io.Serializable` interface to mark it as serializable.
+
+```java
+class Link implements Serializable
+```
+
+We then implement the following method:
+
+```java
+public static void serializeLink(Link inputLink, String fileName) throws IOException {
+
+    FileOutputStream file = new FileOutputStream(fileName);
+    ObjectOutputStream out = new ObjectOutputStream(file);
+    out.writeObject(inputLink);
+    out.close();
+    file.close();
+
+}
+```
+
+#### Description
+
+This function will serialize a `Link` object submitted as a parameter. The serialized bytes are then written to a file in which we will specify the name as the second parameter.  Handle Serialization via `out.writeObject(inputLink)`.  By calling `file.close()`, we terminate the file handle.
+
+#### Deserialization example
+
+The serialized object may now be deserialized by using the technique listed below.
+
+```java
+public static Link deserializeLink(String fileName) throws IOException, ClassNotFoundException {
+
+    FileInputStream file = new FileInputStream(fileName);
+    ObjectInputStream on = new ObjectInputStream(file);
+    return (Link) on.readObject();
+
+}
+```
+
+#### Description
+
+The `deserializeLink()` method has two parameters. The first of which, putoutLink, is a blank `Link` object (which we can initialize earlier, but the original values we replace by this method). The second parameter, filename, is the location where we save the deserialized object.

```

Comment:
The deserializeLink method only has one parameter, that being `fileName` please revise this description.